agent/agent.go

```diff
@@ -1313,6 +1314,39 @@ func (a *Agent) RPC(method string, args interface{}, reply interface{}) error {
 	return a.delegate.RPC(method, args, reply)
 }
 
+// ReconnectSerfWithNewNodeName is used to prepare reconnect to the serf cluster with the new node name
+func (a *Agent) ReconnectSerfWithNewNodeName(oldNodeName string) error {
+	oldMembers := a.delegate.LANMembers()
+
+	if err := a.delegate.ReconnectSerfWithNewNodeName(oldNodeName); err != nil {
+		return err
+	}
+
+	joinAddrs := []string{}
+	for _, m := range oldMembers {
+		// We take 10 addresses out of the pool of known members to start joining the cluster.
+		if len(joinAddrs) < 10 && m.Name != oldNodeName && m.Status == serf.StatusAlive {
+			joinAddrs = append(joinAddrs, fmt.Sprintf("%s:%d", m.Addr.String(), m.Port))
+		}
```

Comment:
You can `break` when `len(joinAddrs) == 10`

Reply:
done